src/app/ui/main_window.cpp

```diff
@@ -90,8 +90,66 @@ MainWindow::MainWindow()
   , m_devConsoleView(nullptr)
 #endif
 {
+}
+
+MainWindow::~MainWindow()
+{
+  delete m_scalePanic;
+
+#ifdef ENABLE_SCRIPTING
+  if (m_devConsoleView) {
+    if (m_devConsoleView->parent() && m_workspace)
+      m_workspace->removeView(m_devConsoleView);
+    delete m_devConsoleView;
+  }
+#endif
+
+  if (m_browserView) {
+    if (m_browserView->parent() && m_workspace)
+      m_workspace->removeView(m_browserView);
+    delete m_browserView;
+  }
+
+  if (m_homeView) {
+    if (m_homeView->parent() && m_workspace)
+      m_workspace->removeView(m_homeView);
+    delete m_homeView;
+  }
+  if (m_contextBar)
+    delete m_contextBar;
+  if (m_previewEditor)
+    delete m_previewEditor;
+
+  // Destroy the workspace first so ~Editor can dettach slots from
+  // ColorBar. TODO this is a terrible hack for slot/signal stuff,
+  // connections should be handle in a better/safer way.
+  if (m_workspace)
+    delete m_workspace;
+
+  // Remove the root-menu from the menu-bar (because the rootmenu
+  // module should destroy it).
+  if (m_menuBar)
+    m_menuBar->setMenu(NULL);
+}
+
+// This 'initialize' function is a way of splitting the creation of
+// the MainWindow. First a minimal instance of MainWindow is made, then
+// all UI components that can trigger the console to report
+// any unexpected errors/warnings are initialized.
+// Prior to this split, Aseprite could crash due to errors during launch.
+// Along the MainWindow creation if any error occurs, whatever it is,
+// Console was called, its constructor tried to create a connection to
+// the MainWindow "Close" command, but resulted in a crash since
+// such an instance did not yet exist.
+// Refer to issue https://github.com/aseprite/aseprite/issues/3914
+// "Fix crash when loading an extension with invalid dithering
+// matrices and the console must be shown"
```

Comment:
Generally we don't add the issue in the comment (as git blame gives this information), but if you want to keep a link to the issue, it can be just:
```
// Refer to https://github.com/aseprite/aseprite/issues/3914
```
(without copying the title)

Anyway probably I'll reword the comment before the merge.

Reply:
I just saw these last comments, I'll take care of making these changes.